File: apps/examples/tenant-xpto/food-market-05-service/src/adapter/in/rest/dto/food-trading.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { FoodTrading } from '../../../../domain/food-trading.entity';

export class FoodTradingDto {
  @ApiProperty()
  id?: number;

  @ApiProperty()
  tradingId: string;

  @ApiProperty()
  marketId: string;

  @ApiProperty()
  itemName: string;

  @ApiProperty()
  quantity: number;

  @ApiProperty()
  unitPrice: number;

  @ApiProperty()
  totalPrice: number;

  @ApiProperty()
  traderName: string;

  @ApiProperty()
  status: string;

  @ApiProperty()
  createdAt: Date;

  static fromDomain(trading: FoodTrading): FoodTradingDto {
    const dto = new FoodTradingDto();
    dto.id = trading.id;
    dto.tradingId = trading.tradingId;
    dto.marketId = trading.marketId;
    dto.itemName = trading.itemName;
    dto.quantity = Number(trading.quantity);
    dto.unitPrice = Number(trading.unitPrice);
    dto.totalPrice = Number(trading.totalPrice);
    dto.traderName = trading.traderName;
    dto.status = trading.status;
    dto.createdAt = trading.createdAt;
    return dto;
  }
}

File: apps/examples/tenant-xpto/food-market-02-service/src/main/kotlin/com/darueira/foodmarket/service02/adapter/in/rest/FoodTradingDto.kt
package com.darueira.foodmarket.service02.adapter.`in`.rest

import com.darueira.foodmarket.service02.domain.model.FoodTrading
import com.fasterxml.jackson.annotation.JsonProperty
import java.math.BigDecimal
import java.time.Instant

data class CreateTradingRequest(
    @JsonProperty("itemName")
    var itemName: String = "",

    @JsonProperty("quantity")
    var quantity: BigDecimal = BigDecimal.ZERO,

    @JsonProperty("unitPrice")
    var unitPrice: BigDecimal = BigDecimal.ZERO,

    @JsonProperty("traderName")
    var traderName: String = ""
)

data class FoodTradingResponse(
    var id: Long? = null,
    var tradingId: String? = null,
    var marketId: String = "",
    var itemName: String = "",
    var quantity: BigDecimal = BigDecimal.ZERO,
    var unitPrice: BigDecimal = BigDecimal.ZERO,
    var totalPrice: BigDecimal? = null,
    var traderName: String = "",
    var status: String = "CONFIRMED",
    var createdAt: Instant = Instant.now()
) {
    companion object {
        fun fromDomain(trading: FoodTrading): FoodTradingResponse {
            return FoodTradingResponse(
                id = trading.id,
                tradingId = trading.tradingId,
                marketId = trading.marketId,
                itemName = trading.itemName,
                quantity = trading.quantity,
                unitPrice = trading.unitPrice,
                totalPrice = trading.totalPrice,
                traderName = trading.traderName,
                status = trading.status,
                createdAt = trading.createdAt
            )
        }
    }
}
